multiversx_sdk/abi/abi_test.py

```diff
@@ -0,0 +1,252 @@
+            self.to = to
+            self.egld_amount = egld_amount
+            self.endpoint_name = endpoint_name
+            self.arguments = arguments
+            self.opt_gas_limit = opt_gas_limit
+
+    encoded_values = abi.encode_endpoint_input_parameters(
+        endpoint_name="proposeBatch",
+        values=[
+            [
+                {
+                    "__discriminant__": 5,
+                    "0": CallActionData(
+                        to=alice,
+                        egld_amount=1000000000000000000,
+                        endpoint_name="example",
+                        arguments=[
+                            bytes([0x03, 0x42]),
+                            bytes([0x07, 0x43])
+                        ],
+                        opt_gas_limit=15_000_000
+                    )
+                }
+            ]
+        ]
+    )
+
+    assert encoded_values == expected_encoded_values
+
+    # Some values typed, structure as dictionary.
+    encoded_values = abi.encode_endpoint_input_parameters(
+        endpoint_name="proposeBatch",
+        values=[
+            [
+                {
+                    "__discriminant__": 5,
+                    "0": {
+                        "to": AddressValue.from_address(alice),
+                        "egld_amount": 1000000000000000000,
+                        "endpoint_name": StringValue("example"),
+                        "arguments": [
+                            bytes([0x03, 0x42]),
+                            bytes([0x07, 0x43])
+                        ],
+                        "opt_gas_limit": U64Value(15_000_000)
+                    }
+                }
+            ]
+        ]
+    )
+
+    # All values typed.
+    encoded_values = abi.encode_endpoint_input_parameters(
+        endpoint_name="proposeBatch",
+        values=[
+            VariadicValues(
+                [
+                    EnumValue(
+                        discriminant=5,
+                        fields=[
+                            Field("0", StructValue([
+                                Field("to", AddressValue.from_address(alice)),
+                                Field("egld_amount", BigUIntValue(1000000000000000000)),
+                                Field("opt_gas_limit", OptionValue(U64Value(15_000_000))),
+                                Field("endpoint_name", BytesValue(b"example")),
+                                Field("arguments", ListValue([
+                                    BytesValue(bytes([0x03, 0x42])),
+                                    BytesValue(bytes([0x07, 0x43])),
+                                ])),
+                            ])),
+                        ],
+                    )
+                ]
+            )
+        ]
+    )
+
+    assert encoded_values == expected_encoded_values
+
+
+def test_decode_endpoint_output_parameters_real_world_multisig_get_pending_action_full_info():
```

Comment:
could have skipped `_real_world_` from the name here and above, but not a big deal :)

Reply:
Indeed, removed (remained in the Go test, for the moment).

https://github.com/multiversx/mx-sdk-abi-go/blob/main/abi/serializer_test.go